Gems/LyShine/Code/Source/UiCanvasComponent.cpp

```diff
@@ -1943,7 +1936,40 @@ void UiCanvasComponent::RenderCanvas(bool isInGame, AZ::Vector2 viewportSize, Ui
     if (m_renderGraph.GetDirtyFlag())
     {
         m_renderGraph.ResetGraph();
+
+        bool renderToTexture = !m_renderInEditor && GetIsRenderToTexture();
+        if (renderToTexture)
+        {
+            if (m_attachmentImageId.IsEmpty())
+            {
+                CreateRenderTarget();
+            }
+
+            // we always clear to transparent black - the accumulation of alpha in the render target requires it
+            AZ::Color clearColor(0.0f, 0.0f, 0.0f, 0.0f);
+
+            // Start building the render to texture node in the render graph
+            AZ::Vector2 viewportTopLeft = AZ::Vector2::CreateZero();
+            AZ::Data::Instance<AZ::RPI::AttachmentImage> attachmentImage;
+            EBUS_EVENT_ID_RESULT(attachmentImage, GetEntityId(), LyShine::RenderToTextureRequestBus, GetRenderTarget, m_attachmentImageId);
```

Comment:
It looks like `UiCanvasComponent` is actually the one that's listening to the `LyShine::RenderToTextureRequestBus` for the `m_entity->GetId()`, so skip the step of using the ebus event and just invoke the `GetRenderTarget` method directly.

Reply:
I actually did that intentionally so that the RenderToTextureRequestBus handler could change without needing to revisit the caller code, but if it's a performance concern it makes sense to make the call directly.

Reply:
I wouldn't expect bus overhead to hit us on this specific case, but it might not hurt to profile this just to see if anything in this code is showing up in a trace (e.g. the UV stuff).

Reply:
OK sounds good. Can you update this to use `EventResult` instead of the old macro-style though?